### PR TITLE
Add 2 new class methods for generating an NxN DFT matrix and its inverse

### PR DESCRIPTION
**Pull request by michaeldzjap**
I just added two class methods for generating an NxN DFT matrix and its inverse: [https://en.wikipedia.org/wiki/DFT_matrix](https://en.wikipedia.org/wiki/DFT_matrix). Would be nice if this could be integrated in the Matrix class.

Thanks.

### CODE REVIEW

**Review by nhthn — APPROVED** [review]
Implementation looks good, but it's up to the maintainers to decide whether they want this feature. Paging @telephon.

File: classes/various/Matrix.sc [hunk 1]
```diff
@@ -44,35 +44,49 @@ Matrix[slot] : Array {
 		});
 		^matrix
 	}
+	*newDFT { arg n;
+		var dft;
+		dft = sqrt(n).reciprocal*super.fill(n, { |r|
```
Comment (telephon) [inline classes/various/Matrix.sc]: could you please add spaces between the operators? That is only a formatting conventions, but still ...

File: classes/various/Matrix.sc [hunk 2]
```diff
@@ -44,35 +44,49 @@ Matrix[slot] : Array {
...
+			Array.fill(n, { |c| Polar(1, -2pi*r*c/n) })
+		});
+		^dft
+	}
+	*newIDFT { arg n;
+		var idft;
+		idft = sqrt(n)*n.reciprocal*super.fill(n, { |r|
+			Array.fill(n, { |c| Polar(1, 2pi*r*c/n) })
```
Comment (telephon) [inline classes/various/Matrix.sc]: same here …

**Review by telephon — CHANGES_REQUESTED** [review]
just minor formatting issues. t would be good if you have an example that could be added to the help files.

**Comment by michaeldzjap** [comment]
I've made the requested modifications. I have also made a new .schelp file for Matrix based on the old .html help file and also added documentation for `*newDFT` and `*newIDFT` in there.

In addition, I have revised the help file for `RootFinder` (I am the original author of that class), so that it can be rendered again. I guess the `MATH::` tag got deprecated somewhere along the line, so just replaced it with `TELETYPE::`. So that open issue can be closed as well.

**Review by telephon — APPROVED** [review]
great! also, thanks a lot for converting the helpfile!

**Pull request merged**